src/category_theory/limits/opposites.lean

```diff
@@ -329,4 +330,201 @@ begin
+end
+
+/-- The canonical isomorphism relating `cospan f.op g.op` and `(span f g).op` -/
+@[simps]
+def cospan_op {X Y Z : C} (f : X ⟶ Y) (g : X ⟶ Z) :
+  cospan f.op g.op ≅ walking_span_op_equiv.inverse ⋙ (span f g).op :=
+nat_iso.of_components (by { rintro (_|_|_); refl, })
+(by { rintros (_|_|_) (_|_|_) f; cases f; tidy, })
+
+/-- The canonical isomorphism relating `(span f g).op` and `cospan f.op g.op` -/
+@[simps]
+def op_span {X Y Z : C} (f : X ⟶ Y) (g : X ⟶ Z) :
+  (span f g).op ≅ walking_span_op_equiv.functor ⋙ cospan f.op g.op :=
+begin
+  calc (span f g).op ≅ 𝟭 _ ⋙ (span f g).op : by refl
+  ... ≅ (walking_span_op_equiv.functor ⋙ walking_span_op_equiv.inverse) ⋙ (span f g).op :
+    iso_whisker_right walking_span_op_equiv.unit_iso _
+  ... ≅ walking_span_op_equiv.functor ⋙ (walking_span_op_equiv.inverse ⋙ (span f g).op) :
+    functor.associator _ _ _
+  ... ≅ walking_span_op_equiv.functor ⋙ cospan f.op g.op :
+    iso_whisker_left _ (cospan_op f g).symm,
+end
+
+namespace pushout_cocone
+
+/-- The obvious map `pushout_cocone f g → pullback_cone f.unop g.unop` -/
+@[simps]
+def unop {X Y Z : Cᵒᵖ} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  pullback_cone f.unop g.unop :=
+cocone.unop ((cocones.precompose (op_cospan f.unop g.unop).hom).obj
+  (cocone.whisker walking_cospan_op_equiv.functor c))
+
+@[simp]
+lemma unop_fst {X Y Z : Cᵒᵖ} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  c.unop.fst = c.inl.unop :=
+by { change (_ : limits.cone _).π.app _ = _, tidy, }
+
+@[simp]
+lemma unop_snd {X Y Z : Cᵒᵖ} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  c.unop.snd = c.inr.unop :=
+by { change (_ : limits.cone _).π.app _ = _, tidy, }
+
+/-- The obvious map `pushout_cocone f.op g.op → pullback_cone f g` -/
+@[simps]
+def op {X Y Z : C} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  pullback_cone f.op g.op :=
+(cones.postcompose ((cospan_op f g).symm).hom).obj
+  (cone.whisker walking_span_op_equiv.inverse (cocone.op c))
+
+@[simp]
+lemma op_fst {X Y Z : C} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  c.op.fst = c.inl.op :=
+by { change (_ : limits.cone _).π.app _ = _, apply category.comp_id, }
+
+@[simp]
+lemma op_snd {X Y Z : C} {f : X ⟶ Y} {g : X ⟶ Z} (c : pushout_cocone f g) :
+  c.op.snd = c.inr.op :=
+by { change (_ : limits.cone _).π.app _ = _, apply category.comp_id, }
+
+end pushout_cocone
+
+namespace pullback_cone
+
+/-- The obvious map `pullback_cone f g → pushout_cocone f.unop g.unop` -/
+@[simps]
+def unop {X Y Z : Cᵒᵖ} {f : X ⟶ Z} {g : Y ⟶ Z} (c : pullback_cone f g) :
+  pushout_cocone f.unop g.unop :=
+cone.unop ((cones.postcompose (op_span f.unop g.unop).symm.hom).obj
+  (cone.whisker walking_span_op_equiv.functor c))
+
+@[simp]
+lemma unop_inl {X Y Z : Cᵒᵖ} {f : X ⟶ Z} {g : Y ⟶ Z} (c : pullback_cone f g) :
+  c.unop.inl = c.fst.unop :=
+begin
+  change ((_ : limits.cocone _).ι.app _) = _,
+  apply quiver.hom.op_inj,
+  tidy,
+  apply category.comp_id,
```

Comment:
It would be nice to get rid of the non-terminal `tidy`. The best I could find is
```suggestion
  change ((_ : limits.cocone _).ι.app _) = _,
  dsimp only [unop, op_span],
  simp, dsimp, simp, dsimp, simp
```
which isn't great either, but at least the horribleness is at the end of the proof.